docs/src/main/asciidoc/security-jpa-concept.adoc

```diff
@@ -0,0 +1,120 @@
+
+* Each `String` role element type is parsed as a comma-separated list of roles.
+
+== JPA entity as storage of roles
+
+Use the following example to store roles inside another JPA entity:
+
+[source,java]
+----
+@UserDefinition
+@Table(name = "test_user")
+@Entity
+public class User extends PanacheEntity {
+    @Username
+    public String name;
+
+    @Password
+    public String pass;
+
+    @ManyToMany
+    @Roles
+    public List<Role> roles = new ArrayList<>();
+}
+
+@Entity
+public class Role extends PanacheEntity {
+
+    @ManyToMany(mappedBy = "roles")
+    public List<ExternalRolesUserEntity> users;
+
+    @RolesValue
+    public String role;
+}
+----
+
+== Password storage and hashing
+
+Regarding password storage and hashing management from a developer perspective, you can decide if you want to keep the settings as it is and use the Quarkus default or hash passwords manually.
+
+With the default option, passwords are stored and hashed with https://en.wikipedia.org/wiki/Bcrypt[bcrypt] under the
+https://en.wikipedia.org/wiki/Crypt_(C)[Modular Crypt Format] (MCF).
+While using MCF, the hashing algorithm, iteration count, and salt are stored as a part of the hashed value.
+As such, we do not need dedicated columns to keep them.
+
+[NOTE]
+====
+In cryptography, a salt is a name for random data used as an additional input to a one-way function that hashes data, a password, or a passphrase.
+====
+
+For manual password hashing, create a class that implements the `CustomPasswordProvider`as shown in the example below.
+
+This is an example of providing a custom password provider using the SHA256 hashing algorithm.
+
+[source,java]
+----
+@UserDefinition
+@Table(name = "test_user")
+@Entity
+public class CustomPasswordUserEntity {
+    @Id
+    @GeneratedValue
+    public Long id;
+
+    @Column(name = "username")
+    @Username
+    public String name;
+
+    @Column(name = "password")
+    @Password(value = PasswordType.CUSTOM, provider = CustomPasswordProvider.class)
+    public String pass;
+
+    @Roles
+    public String role;
+}
+
+public class CustomPasswordProvider implements PasswordProvider {
+    @Override
+    public Password getPassword(String pass) {
+        byte[] digest = DatatypeConverter.parseHexBinary(pass);
+        return SimpleDigestPassword.createRaw(SimpleDigestPassword.ALGORITHM_SIMPLE_DIGEST_SHA_256, digest);
+    }
+}
+----
+
+[TIP]
+====
+For quick creation of a hashed password, use `String BcryptUtil.bcryptHash(String password)`, which defaults to creating a random salt and hashing in ten iterations.
+This method also allows specifying the desired amount of iterations and the salt used.
+====
+
+[WARNING]
+====
+Never store passwords as a plain text for applications already running in production.
+
+However, it is possible to store passwords as a plain text with the `@Password(PasswordType.CLEAR)` annotation when operating in a test environment.
+====
```

Comment:
Please add Reeferences section here, every sec doc should link back to the main security overview concept and other related concept for the ease of navigation

Reply:
Added ;) Good idea